src/main/java/org/sonarlint/intellij/issue/persistence/LiveIssueCache.java

```diff
@@ -87,7 +85,10 @@ protected boolean removeEldestEntry(Map.Entry<VirtualFile, Collection<LiveIssue>
    */
   @CheckForNull
   public synchronized Collection<LiveIssue> getLive(VirtualFile virtualFile) {
-    return cache.get(virtualFile);
+    if(cache.get(virtualFile) == null) {
```

Comment:
Same here, even if it is synchronized, I would change the code to be more concurrent friendly. Please also ensure you follow code formatting conventions